src/DocumentFormat.OpenXml/Wordprocessing/Document.cs

```diff
@@ -3,20 +3,122 @@
 
 using DocumentFormat.OpenXml.Framework;
 using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
 
 namespace DocumentFormat.OpenXml.Wordprocessing
 {
     public partial class Document
     {
+        private static readonly RNGCryptoServiceProvider Generator = new();
```

Comment:
Can create a service that abstracts these things? It should handle the generation, the duplicate checking, etc

Reply:
What exactly do you have in mind? How should `Paragraph` and `TableRow` instances access that service? Wouldn't that add a potentially unnecessary layer?

Why not have the `Document` instance provide that service? It defines the scope within which `w14:paraId` values must be unique. Thus, we'd need one service instance per `Document` instance anyhow.

Reply:
The Document instance can provide it, but the implementation for it would be much easier to test if it were in its own service. The Document class (and all the strongly-typed classes provided) should have limited additional logic.